packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -356,56 +363,85 @@ export class TransformerResolver implements TransformerResolverProvider {
-            dataSource = `$util.qr($ctx.stash.metadata.put("databaseName", "${databaseName}"))`;
+            dataSource = stashStatement('databaseName', `"${databaseName}"`);
           }
           break;
         default:
           throw new Error('Unknown DataSource type');
       }
     }
     let initResolver = dedent`
-      $util.qr($ctx.stash.put("typeName", "${this.typeName}"))
-      $util.qr($ctx.stash.put("fieldName", "${this.fieldName}"))
-      $util.qr($ctx.stash.put("conditions", []))
-      $util.qr($ctx.stash.put("metadata", {}))
-      $util.qr($ctx.stash.metadata.put("dataSourceType", "${dataSourceType}"))
-      $util.qr($ctx.stash.metadata.put("apiId", "${api.apiId}"))
-      $util.qr($ctx.stash.put("connectionAttributes", {}))
+      ${stashStatement('typeName', `"${this.typeName}"`)}
+      ${stashStatement('fieldName', `"${this.fieldName}"`)}
+      ${stashStatement('conditions', '[]')}
+      ${stashStatement('metadata', '{}')}
+      ${stashStatement('dataSourceType', `"${dataSourceType}"`, 'metadata')}
+      ${stashStatement('apiId', `"${api.apiId}"`, 'metadata')}
+      ${stashStatement('connectionAttributes', '{}')}
       ${dataSource}
     `;
     const account = Stack.of(context.stackManager.scope).account;
     const authRole = context.synthParameters.authenticatedUserRoleName;
     if (authRole) {
+      const authRoleStatement = stashStatement('authRole', `"arn:aws:sts::${account}:assumed-role/${authRole}/CognitoIdentityCredentials"`);
+
       initResolver += dedent`\n
-      $util.qr($ctx.stash.put("authRole", "arn:aws:sts::${account}:assumed-role/${authRole}/CognitoIdentityCredentials"))
+        ${authRoleStatement}
       `;
     }
     const unauthRole = context.synthParameters.unauthenticatedUserRoleName;
     if (unauthRole) {
+      const unauthRoleStatement = stashStatement(
+        'unauthRole',
+        `"arn:aws:sts::${account}:assumed-role/${unauthRole}/CognitoIdentityCredentials"`,
+      );
       initResolver += dedent`\n
-      $util.qr($ctx.stash.put("unauthRole", "arn:aws:sts::${account}:assumed-role/${unauthRole}/CognitoIdentityCredentials"))
+        ${unauthRoleStatement}
       `;
     }
     const identityPoolId = context.synthParameters.identityPoolId;
     if (identityPoolId) {
+      const identityPoolStatement = stashStatement('identityPoolId', `"${identityPoolId}"`);
       initResolver += dedent`\n
-        $util.qr($ctx.stash.put("identityPoolId", "${identityPoolId}"))
+        ${identityPoolStatement}
       `;
     }
     const adminRoles = context.synthParameters.adminRoles ?? [];
+    const adminRolesStatement = stashStatement('adminRoles', `${JSON.stringify(adminRoles)}`);
     initResolver += dedent`\n
-      $util.qr($ctx.stash.put("adminRoles", ${JSON.stringify(adminRoles)}))
+      ${adminRolesStatement}
     `;
-    initResolver += '\n$util.toJson({})';
+
+    if (this.runtime?.name === 'APPSYNC_JS') {
```

Comment:
Here and throughout: can we do something other than a magic string comparisons for runtime evaluation? E.g., centralize a constant, add a type predicate, etc

Reply:
Good call, addressed in [113d921](https://github.com/aws-amplify/amplify-category-api/pull/2812/commits/113d921c23ec8d9262af4a05dca7705b881dcb82)